### PR TITLE
dev/core#4551 Fix escaping on token event.title

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
See https://lab.civicrm.org/dev/core/-/issues/4551

Before
----------------------------------------
Fatal on receipting with single quote in event name

After
----------------------------------------
Fixed

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Core/EntityTokens.php [hunk 1]
```diff
@@ -688,4 +688,15 @@ protected function getCacheKey(): string {
     return $cacheKey;
   }
 
+  /**
```
Comment (larssandergreen) [inline CRM/Core/EntityTokens.php:691]: This function already exists at 154

Comment (eileenmcnaughton) [reply CRM/Core/EntityTokens.php:691]: isn't that only in master?

Comment (larssandergreen) [reply CRM/Core/EntityTokens.php:691]: Right, didn't notice this was 5.65

Comment (eileenmcnaughton) [reply CRM/Core/EntityTokens.php:691]: @larssandergreen it will merge-conflict when we try to up-merge but ....

**Comment by larssandergreen** [comment]
This fixes the problem with the quotes not being escaped, but now the html fields (e.g. pay later receipt, though you need to [fix the template to see it](https://github.com/civicrm/civicrm-core/pull/27335)) are coming through with everything rendered as html entities — so messed up.

**Comment by larssandergreen** [comment]
Or rather I should say they are coming through as text, but with html entities in them (the raw content is like `&lt;p&gt;&lt;a href=&quot;`).

**Comment by larssandergreen** [comment]
I see the problem, none of this will work right without the other recent changes. I think since this is an edge case (and seems to work sometimes anyways), maybe it would be better to just target master with this.

**Comment by eileenmcnaughton** [comment]
@larssandergreen this is the nl2br & br2nl changes?

**Comment by larssandergreen** [comment]
@eileenmcnaughton Yes. I tested this change on master instead and it works as intended and doesn't cause any issues with the html fields.

**Comment by eileenmcnaughton** [comment]
@larssandergreen I think the problem is just the pay_later_receipt field metadata doesn't specify html - so I pushed up a schema definition

Note that lots of cache clearing needed

**Comment by larssandergreen** [comment]
Yes, that was it. I was getting tripped up by the cache. All good now.

**Comment by seamuslee001** [comment]
I'm going to merge based on Lars's testing so we can get it into 5.65.0

**Pull request merged**